deepinv/tests/test_metric.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+import torch
+from deepinv.physics import GaussianNoise
+import deepinv.loss.metric as metric
+from deepinv.tests.dummy_datasets.datasets import DummyCircles
+from deepinv.utils.demo import get_image_url, load_url_image
+
+METRICS = ["MAE", "MSE", "NMSE", "PSNR", "SSIM", "LpNorm", "L1L2", "LPIPS", "NIQE"]
+FUNCTIONALS = ["cal_mse", "cal_mae", "cal_psnr"]
+
+
+def choose_metric(metric_name, **kwargs) -> metric.Metric:
+    if metric_name in ("LPIPS", "NIQE"):
+        pytest.importorskip(
+            "pyiqa",
+            reason="This test requires pyiqa. It should be "
+            "installed with `pip install pyiqa`",
+        )
+    if metric_name == "MSE":
+        return metric.MSE(**kwargs)
+    elif metric_name == "NMSE":
+        return metric.NMSE(**kwargs)
+    elif metric_name == "MAE":
+        return metric.MAE(**kwargs)
+    elif metric_name == "PSNR":
+        return metric.PSNR(**kwargs)
+    elif metric_name == "SSIM":
+        return metric.SSIM(**kwargs)
+    elif metric_name == "LpNorm":
+        return metric.LpNorm(**kwargs)
+    elif metric_name == "L1L2":
+        return metric.L1L2(**kwargs)
+    elif metric_name == "LPIPS":
+        return metric.LPIPS(**kwargs)
+    elif metric_name == "NIQE":
+        return metric.NIQE(**kwargs)
+
+
+@pytest.mark.parametrize("metric_name", METRICS)
+@pytest.mark.parametrize("complex_abs", [True])
+@pytest.mark.parametrize("train_loss", [True, False])
+@pytest.mark.parametrize("norm_inputs", [None])
+def test_metrics(metric_name, complex_abs, train_loss, norm_inputs, rng):
+    m = choose_metric(
+        metric_name,
+        complex_abs=complex_abs,
+        train_loss=train_loss,
+        norm_inputs=norm_inputs,
+        reduction="mean",
+    )
+    x = load_url_image(
+        get_image_url("celeba_example.jpg"), img_size=128, resize_mode="resize"
+    )
+
+    if complex_abs:
+        x = x[:, :2, ...]
+
+    x_hat = GaussianNoise(sigma=0.1, rng=rng)(x)
+
+    # Test metric worse when image worse
+    if train_loss:  # All metrics lower = better
```

Comment:
some metrics can have higher=better, right?

Reply:
i've added a comment to explain this in the test